From: Mike Christie <mchristi@redhat.com>
Date: Tue, 5 May 2009 15:27:55 -0500
Subject: [scsi] fnic: init retry counter
Message-id: 4A00A14B.6000607@redhat.com
O-Subject: Re: [PATCH 2/3] RHEL 5.4: update fnic fcoe driver
Bugzilla: 484438

mchristi@redhat.com wrote:
> From: Mike Christie <mchristi@redhat.com>
>
> This is also for BZ 484438. fnic is the primary user of
> libfc.
>
> This syncs the driver with what the scsi maintainer is going
> to take for upstream (James Bottomley said he was ok with
> it but has not got a chance to merge it yet).
>
> I have only compile tested this. Cisco is doing all the
> testing.

There was a bug found in this patch where fnic was not initializing a
retry counter. I have attached a incremental patch,
set-lport-retry-count-and-ver.patch, and I have attached a updated
complete patch 0001-RHEL-5.4-update-fnic-fcoe-driver-v2.patch. So
reviewers can check out the incremental and Don can merge whichever is
easier.

>From 865064150295997fcb26801ba9d99b9a507073cb Mon Sep 17 00:00:00 2001
From: Mike Christie <mchristi@redhat.com>
Date: Tue, 5 May 2009 14:21:54 -0500
Subject: [PATCH] RHEL 5.4: update fnic fcoe driver v2

This is also for BZ 484438. fnic is the primary user of
libfc.

This syncs the driver with what the scsi maintainer is going
to take for upstream (James Bottomley said he was ok with
it but has not got a chance to merge it yet).

I have only compile tested this. Cisco is doing all the
testing.

v2
- set lport retry counter

diff --git a/drivers/scsi/fnic/fnic.h b/drivers/scsi/fnic/fnic.h
index d55e5e3..21e4f5f 100644
--- a/drivers/scsi/fnic/fnic.h
+++ b/drivers/scsi/fnic/fnic.h
@@ -35,7 +35,7 @@
 
 #define DRV_NAME		"fnic"
 #define DRV_DESCRIPTION		"Cisco FCoE HBA Driver"
-#define DRV_VERSION		"1.0.0.1121"
+#define DRV_VERSION		"1.0.0.1122"
 #define PFX			DRV_NAME ": "
 #define DFX                     DRV_NAME "%d: "
 
diff --git a/drivers/scsi/fnic/fnic_main.c b/drivers/scsi/fnic/fnic_main.c
index d25ea0f..fd3a119 100644
--- a/drivers/scsi/fnic/fnic_main.c
+++ b/drivers/scsi/fnic/fnic_main.c
@@ -690,6 +690,7 @@ static int __devinit fnic_probe(struct pci_dev *pdev,
 	}
 
 	lp->max_retry_count = fnic->config.flogi_retries;
+	lp->max_rport_retry_count = fnic->config.plogi_retries;
 	lp->service_params = (FCP_SPPF_INIT_FCN | FCP_SPPF_RD_XRDY_DIS |
 			      FCP_SPPF_CONF_COMPL);
 	if (fnic->config.flags & VFCF_FCP_SEQ_LVL_ERR)
